drivers/interrupt_controller/intc_gicv3_priv.h

```diff
@@ -266,4 +266,16 @@ void its_rdist_invall(void);
 extern atomic_t nlpi_intid;
 #endif
 
+#if defined(CONFIG_ARMV8_A_NS) || defined(CONFIG_GIC_SINGLE_SECURITY_STATE)
+static ALWAYS_INLINE void sys_write_irouter(uint64_t val, mem_addr_t addr)
```

Comment:
If any, this function should be named `arm_gic_irouter`.

Also this function in itself does not seem to have anything to do with the `IROUTER`, which makes me wonder if the `irouter` part should be dropped from its name.

Reply:
seems all sys_read/write64 in gicv3 has this issue with aarch32.
it's `arm_gic_write64` better?

Reply:
> it's `arm_gic_write64` better?

something like that would make more sense IMHO.

Reply:
Agree, i updated